### ISSUE TITLE
classic theme: inconsistent padding in admonitions vs. topic/sidebar

### ISSUE THREAD

**Issue by mgeier** [sphinx-doc/sphinx#7544]
Admonitions have a padding of `7px` on all 4 sides (defined in `classic.css`), while `topic` and `sidebar` have a padding of `7px` only on the left, right and top, not on the bottom (defined in `basic.css`).

Is this intentional?

If yes, we should probably remove the bottom padding from `classic.css`, if no, we should probably add the padding in `basic.css`?

**Comment by tk0miya** [comment]
Sorry, I don't know. It was created 10+ years ago...

**Comment by mgeier** [comment]
Is there anybody we could ask?

This is clearly an inconsistency (between admonitions on one hand and topic and sidebar on the other hand) in the `classic`, `haiku` and `nature` themes.

I don't think it is a problem with the other built-in themes, most notably the `sphinxdoc` theme, which uses `padding: 0` in admonitions (which has it's own problems, but I don't want to go too deep into that here).

**Comment by tk0miya** [comment]
I think nobody can't answer your question. But that is not a reason not to fix the padding. I'm not good at CSS. But I agree it's strange the 0-paddings. So let's fix it.

**Comment by mgeier** [comment]
I'm not sure what's the best way to fix this.

The 0-paddings are strange, but I think they are intentional.

I've created a little example to show the problem (using the `classic` theme):

```rst
.. note::

   .. warning::

      Warning in note.

.. topic:: Topic:

   .. warning::

      Warning in topic.
```

![image](https://user-images.githubusercontent.com/705404/81077510-5b659a00-8eed-11ea-9cad-17163463657e.png)

The space below the warning in the note is `10px` margin (of the warning) plus `7px` padding (of the note).

The space below the warning in the topic is `10px` margin (of the warning) plus nothing, because the topic has no `padding-bottom`.

Arguably, both are wrong, because I would expect a space of `7px` on all sides.

I think the "right" solution would be to remove the `margin-bottom` from the `:last-child` and add a consistent `padding: 7px` on all sides.
The problem is that this would be quite a big change in multiple themes, and people might consider it a "breaking change".

> I'm not good at CSS.

Me neither. But I'm learning ...

It would be great to hear what a CSS expert can say about this situation.

**Comment by mgeier** [comment]
Related issue in the RTD theme: https://github.com/readthedocs/sphinx_rtd_theme/issues/803.

**Comment by mgeier** [comment]
I've created a PR trying to fix this situation: #7657.
